iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/topology/RemoveHvdcLineTest.java

```diff
@@ -0,0 +1,117 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.powsybl.iidm.modification.topology.TopologyTestUtils.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author Anis Touri <anis-1.touri@rte-france.com>
+ */
+class RemoveHvdcLineTest extends AbstractConverterTest {
+
+    private final Set<String> removedObjects = new HashSet<>();
+    private final Set<String> beforeRemovalObjects = new HashSet<>();
+
+    @AfterEach
+    public void tearDown() {
+        removedObjects.clear();
+    }
+
+    private void addListener(Network network) {
+        network.addListener(new DefaultNetworkListener() {
+            @Override
+            public void beforeRemoval(Identifiable id) {
+                beforeRemovalObjects.add(id.getId());
+            }
+
+            @Override
+            public void afterRemoval(String id) {
+                removedObjects.add(id);
+            }
+        });
+    }
+
+    @Test
+    void testRemoveHvdcLineLcc() {
+        Network network = HvdcTestNetwork.createLcc();
+        addListener(network);
+        new RemoveHvdcLineBuilder().withHvdcLineId("L", null).build().apply(network);
+        assertEquals(Set.of("L", "C1", "C2"), beforeRemovalObjects);
+        assertEquals(Set.of("L", "C1", "C2"), removedObjects);
+        assertNull(network.getHvdcLine("L"));
+        assertNull(network.getLccConverterStation("C1"));
+        assertNotNull(network.getShuntCompensator("C1_Filter1"));
+    }
+
+    @Test
+    void testRemoveHvdcLineLccWithMcs() {
+        Network network = HvdcTestNetwork.createLcc();
+        addListener(network);
+        new RemoveHvdcLineBuilder().withHvdcLineId("L", List.of("C1_Filter1")).build().apply(network);
+        assertEquals(Set.of("L", "C1", "C2", "C1_Filter1"), beforeRemovalObjects);
+        assertEquals(Set.of("L", "C1", "C2", "C1_Filter1"), removedObjects);
+        assertNull(network.getHvdcLine("L"));
+        assertNull(network.getLccConverterStation("C1"));
+        assertNull(network.getShuntCompensator("C1_Filter1"));
+    }
+
+    @Test
+    void testRemoveHvdcLineVsc() {
+        Network network = HvdcTestNetwork.createVsc();
+        addListener(network);
+        new RemoveHvdcLineBuilder().withHvdcLineId("L", null).build().apply(network);
+        assertEquals(Set.of("L", "C1", "C2"), beforeRemovalObjects);
+        assertEquals(Set.of("L", "C1", "C2"), removedObjects);
+        assertNull(network.getHvdcLine("L"));
+        assertNull(network.getVscConverterStation("C1"));
+        assertNull(network.getVscConverterStation("C2"));
+    }
+
+    @Test
+    void testRemoveHvdcWithLccConverterStation() throws IOException {
+        Network network = HvdcTestNetwork.createLcc();
+        NetworkModification modification = new RemoveHvdcLineBuilder().withHvdcLineId("L", null).build();
+        modification.apply(network);
+        roundTripXmlTest(network, NetworkXml::writeAndValidate, NetworkXml::validateAndRead,
```

Comment:
You don't need xiidm based unit test/round trips. This already tested somewhere else. Assertions on Java model is enough.